airflow/providers/mongo/hooks/mongo.py

```diff
@@ -75,6 +74,10 @@ def __init__(self, mongo_conn_id: str = default_conn_name, *args, **kwargs) -> N
         self.client: MongoClient | None = None
         self.uri = self._create_uri()
 
+        self.allow_insecure = True
+        if self.extras.get("ssl", False):
```

Comment:
According to the [`MongoClient`](https://pymongo.readthedocs.io/en/stable/api/pymongo/mongo_client.html#pymongo.mongo_client.MongoClient) `ssl` is alias to the `tls`, potentially end users might use `tls` instead of `ssl`

Reply:
Should I integrate it here or take it in a followup? Addressing a vulnerability here